apps/server/src/toasts/toast.service.ts

```diff
@@ -2,15 +2,173 @@ import { Injectable } from '@nestjs/common';
+    const firstDateForPeriod = new Date(currentDate.getFullYear(), 0, 1);
+    const secondDateForPeriod = new Date(currentDate.getFullYear(), 6, 1);
+    const secondEndDateForPeriod = new Date(
+      currentDate.getFullYear() + 1,
+      0,
+      1
+    );
+
+    if (currentDate.getMonth() + 1 < 7) {
+      return this.toastModel.count({
+        where: {
+          toastDate: {
+            [Op.gt]: firstDateForPeriod,
+            [Op.lt]: secondDateForPeriod,
+          },
+        },
+      });
+    } else {
+      return this.toastModel.count({
+        where: {
+          toastDate: {
+            [Op.gt]: secondDateForPeriod,
+            [Op.lt]: secondEndDateForPeriod,
+          },
+        },
+      });
+    }
+  }
+
+  getAllTimeRecord() {
+    // return this.toastModel.count({attributes:["toastDate" , [Sequelize.fn('CASE' ) , sequelize.  ]]})}
+
+    //Sequelize.when(Sequelize.fn('MONTH', Sequelize.col('toastDate')) , {$lte: 6})
 
+    const firstHalfPromise = this.toastModel.findAll({
+      attributes: [
+        [Sequelize.fn('COUNT', Sequelize.col('toastDate')), 'amountOfToasts'],
+        [
+          Sequelize.fn('DATE_TRUNC', 'year', Sequelize.col('toastDate')),
+          'date',
+        ],
+      ],
+
+      group: [
+        Sequelize.fn('DATE_TRUNC', 'year', Sequelize.col('toastDate')),
+        'date',
+      ],
+      order: [[Sequelize.fn('COUNT', Sequelize.col('toastDate')), 'DESC']],
+      where: Sequelize.where(
+        Sequelize.fn('date_part', 'month', Sequelize.col('toastDate')),
+        {
+          [Op.lte]: 6,
+        }
+      ),
+      limit: 1,
+    });
+    const secondHalfPromise = this.toastModel.findAll({
+      attributes: [
+        [Sequelize.fn('COUNT', Sequelize.col('toastDate')), 'amountOfToasts'],
+        [
+          Sequelize.fn('DATE_TRUNC', 'year', Sequelize.col('toastDate')),
+          'date',
+        ],
+      ],
+
+      group: ['toastDate'],
+      order: [[Sequelize.fn('COUNT', Sequelize.col('toastDate')), 'DESC']],
+      where: Sequelize.where(
+        Sequelize.fn('date_part', 'month', Sequelize.col('toastDate')),
+        {
+          [Op.gt]: 6,
+        }
+      ),
+      limit: 1,
+    });
+    const biggerOne = Promise.all([firstHalfPromise, secondHalfPromise]).then(
+      (values) => {
+        return values[0][0].dataValues['amountOfToasts'] >
+          values[1][0].dataValues['amountOfToasts']
+          ? values[0][0].dataValues['amountOfToasts']
+          : values[1][0].dataValues['amountOfToasts'];
+      }
+    );
+
+    return biggerOne;
+  }
+
+  getAmountToastsForCurrentPeriodPerUser() {
+    const currentDate = new Date();
+    const firstDateForPeriod = new Date(currentDate.getFullYear(), 0, 1);
+    const secondDateForPeriod = new Date(currentDate.getFullYear(), 6, 1);
+    const secondEndDateForPeriod = new Date(
+      currentDate.getFullYear() + 1,
+      0,
+      1
+    );
+
+    if (currentDate.getMonth() + 1 < 7) {
```

Comment:
avoid magic numbers. 7 is unclear.
also. adding 1 isn't clear. I suggest you use the date-fns library to avoid the one off issue with regular js months